source/end_sixtrack.f90

```diff
@@ -111,125 +111,61 @@ subroutine abend(cstring)
   sumda(:) = zero
   call time_timerCheck(time1)
   trtime = time1 - time0
-#ifdef CR
   trtime = trtime + crtime3
-#endif
   sumda(60) = real(trtime,fPrec)  ! Track time
   sumda(52) = real(numvers,fPrec) ! SixTrack version
+  outLine = " "
+  k = 1
+  do i=1,60
+    call chr_fromReal(sumda(i),chBuf,19,2,rErr)
+    outLine(k:k+25)=" "//chBuf(1:25)
+    k = k+26
+  end do
 
   ! Note it COULD happen that napxo is 0 for a very very early error and even napx!!!
   if(napxo == 0 .and. napx == 0) napxo = 1
   if(napxo == 0) napxo = napx
   write(93,"(2(a,i0))") "SIXTRACR> STOP/ABEND writing fort.10, lines ",napxo,"/",napx
   flush(93)
-  do j=1,napxo
-    l1 = 1
-    do i=1,60
-      call chr_fromReal(sumda(i),ch1,19,2,rErr)
-      ch(l1:l1+25)=" "//ch1(1:25)
-      l1=l1+26
-    end do
-    write(10,"(a)",iostat=ierro) ch(1:l1-1)
-    if(ierro /= 0) then
-      write(lout,"(a,i0)") "ABEND> ERROR Problems writing to fort.10. ierro = ",ierro
-    end if
+  do j=1,napxo ! Must the dummy file really be napxo times the same dummy line?
+    write(10,"(a)",iostat=ierro) outLine(1:60*26)
   end do
+  if(ierro /= 0) then
+    write(lerr,"(a,i0)") "ABEND> ERROR Problems writing to fort.10. ierro = ",ierro
+  end if
+#endif
 
 12 continue
-  call f_close(10)
-
-#ifdef CR
-  call f_close(91)
-  call f_close(95)
-  call f_close(96)
   if(lout == 92) then
     write(93,"(a)") "SIXTRACR> STOP/ABEND copying fort.92 to fort.6"
     flush(93)
     rewind(92)
-3   read(92,'(a1024)',end=1,err=8,iostat=ierro) arecord
-    lstring=1024
-    do i=1024,2,-1
-      lstring=i
-      if(arecord(i:i) /= " ") goto 2
-      lstring=lstring-1
-    end do
-2   write(6,'(a)',iostat=ierro) arecord(1:lstring)
-    goto 3
+13  continue
+    read(92,"(a)",err=14,end=15,iostat=ierro) inLine
+    write(6,'(a)',iostat=ierro) trim(inLine)
+    goto 13
   end if
 
-1 write(6,"(a)",iostat=ierro) "SIXTRACR> Stop "//cstring
-  ! call f_close(6)
+14 continue
+  write(93,"(a,i0)")   "SIXTRACR> ERROR Reading fort.92 in abend, iostat = ",ierro
+  write(lerr,"(a,i0)") "ABEND> ERROR Reading fort.92, iostat = ",ierro
+
+15 continue
+  write(6,"(a)",iostat=ierro) "SIXTRACR> Stop "//cstring
   rewind(92) ! Get rid of fort.92
   endfile(92,iostat=ierro)
   call f_close(92)
   write(93,"(a)") "SIXTRACR> Stop "//cstring
+  call f_close(93)
+
 #ifdef BOINC
```

Comment:
main_cr? Don't know. Not here.

Reply:
Hmm. At some point I made sure that zipf got done at the very end of a normal exit, after closing all the files and writing the final remarks to fort.6. But I don't think it should be done at all at for an abnormal exit, since the sky may be falling and we want to close down ASAP not go wandering into C++-land...

Something to verify...